include/gmock-win32.h

```diff
@@ -804,12 +804,14 @@ struct mock_module_##func \
 #define MOCK_MODULE_SUB_NBARG(...) MOCK_MODULE_EXPAND(MOCK_MODULE_LASTOF15(__VA_ARGS__, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0))
 #define MOCK_MODULE_NBARG(...) MOCK_MODULE_SUB_NBARG(MOCK_MODULE_PREFIX(__VA_ARGS__))
 
-#define MOCK_MODULE_OVERLOAD(name, count) MOCK_MODULE_CONCAT(name, count)
+#define MOCK_MODULE_OVERLOAD(count_args, conv, signat) MOCK_MODULE_MAKENAME_( count_args, conv )signat
+// Expand before concatenate
+#define MOCK_MODULE_MAKENAME_(count_args, conv) MOCK_MODULE_FUNC##count_args##conv
 #define MOCK_MODULE_FUNC_OVERLOAD(name, ...) MOCK_MODULE_UNITE(MOCK_MODULE_OVERLOAD(name, MOCK_MODULE_NBARG(__VA_ARGS__)), (__VA_ARGS__))
```

Comment:
It seems like the `MOCK_MODULE_FUNC_OVERLOAD` macro isn't being used anymore.

Reply:
Unfortunately no.

For example:
if rewrite MOCK_MODULE_OVERLOAD to MOCK_MODULE_OVERLOAD2~=MOCK_MODULE_MAKENAME_
like this
`#define MOCK_MODULE_OVERLOAD2(count_args, conv, signat) MOCK_MODULE_FUNC##count_args##conv signat`

For code
`MOCK_MODULE_FUNC( UINT, GetACP );`
results in wrong code:
`MOCK_MODULE_FUNCMOCK_MODULE_NBARG()(GetACP,UINT()))`

So you need someone to "evaluate" `MOCK_MODULE_NBARG()` to expand it to
`MOCK_MODULE_FUNC##0##conv signat`
Putting It All Together:
`MOCK_MODULE_OVERLOAD` take `MOCK_MODULE_NBARG()` and prepare it to use in helper `MOCK_MODULE_MAKENAME_`
More [SO, How can I concatenate twice with the C preprocessor and expand a macro as in "arg ## _ ## MACRO"?](https://stackoverflow.com/questions/1489932/how-can-i-concatenate-twice-with-the-c-preprocessor-and-expand-a-macro-as-in-ar)

__Macro still usefull.__
*There will be two more new improvement macros soon, I didn’t want to post everything at once.*